lib/mongo/retryable.rb

```diff
@@ -296,12 +296,12 @@ def modern_read_with_retry(session, server_selector, &block)
       begin
         yield server
       rescue Error::SocketError, Error::SocketTimeoutError => e
-        if session.in_transaction?
+        if session && session.in_transaction?
```

Comment:
`modern_read_with_retry` is only called from line 117 in this file, which is the branch of an if statement conditional on session being present.